### PR TITLE
Feature/jcs location comp

### LINKED ISSUE #51: Location Detail Page Styling for Dynamic Data
- [ ] Refactor for detail displaying dynamically
- [ ] Readable scss 
- [ ] Follow ux of project

### LINKED ISSUE #52: WeeklyForcast Component
- [ ] add component to map the 7 day forcast to display each upcoming day
- [ ] match structure of Location components jsx
- [ ] Add types to all props being passed in

### LINKED ISSUE #53: Typescript Remove any types
- [ ] work on removing any instances of the any type from the project
- [ ] This will be an on going project 
- [ ] Get used to the errors and what they mean

### PR DESCRIPTION
**Pull request by JoshSevy**
# Description
- Worked with removing type any where possible.
- Lots of issues trying to fetch data in Location component so moved to App
- Lots of typescript errors I had to walk through --- this is what ate up most of my time yesterday
- Location is displaying dynamic data from the search results page
- Merged from master so no merge conflicts on this end.
- Browser router can cause looping of fetch calls in components --- lots of research to where this was happening

## Type of change
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other, Please explain:
`Your message here`
## Has This Been Tested?
- [x] No
- [x] Yes
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration:

# Message for reviewer:
That was a rough path, lots of TS errors to work through. Sorry I didn't anticipate it to take as long as it did.
# Checklist:
- [x] My code has no unused/commented out code
--- WeeklyForcast component is commented out until I can get the TS errors corrected
- [x] My code has no console logs
- [x] I have performed a self-review of my own code
- [x] I have commented on my code, particularly in hard-to-understand areas --- In progress
Screen Shot 2020-09-13 at 11 44 04 AM
Screen Shot 2020-09-13 at 11 43 41 AM

### CODE REVIEW

File: src/App/App.tsx [hunk 1]
```diff
@@ -8,24 +8,33 @@ import Location from "../Location/Location"
...
+import { getAllData } from "../helpers/dataFilter"
+import { DetailsProps } from '../helpers/detailsdefinitions'
 
 
-const App: React.FC = () => {
+const App: React.FC<DetailsProps> = () => {
   const [ searchResults, setSearchResults ] = useState([])
-  const [ query, setQuery ] = useState('')
-	const [ matchDetails, setMatchDetails ] = useState<Array<any>>([])
+  const [ query, setQuery ] = useState<string>()
```
Comment (ErinUntermeyer) [inline src/App/App.tsx:17]: Thanks for cleaning up the TS here! It looks great.

File: src/App/App.tsx [hunk 2]
```diff
@@ -8,24 +8,33 @@ import Location from "../Location/Location"
...
 	}
 	
-	const getMatchDetails = (coordinates: [], locality: string, region: string) => {
-		setMatchDetails([coordinates, locality, region])
-	}
+	const getMatchDetails = (latitude: number, longitude: number, locality: string, region: string) => {
+		setMatchDetails([latitude, longitude, locality, region])
+  }
+
+  const getAllDetailsData = async (lat: number, long: number) => {
```
Comment (ErinUntermeyer) [inline src/App/App.tsx:33]: I'm happy it worked out fetching the data here. What was going on with fetching it in Location? I'm curious!

File: src/Location/Location.tsx [hunk 3]
```diff
@@ -1,148 +1,129 @@
...
-interface LocationProps {
-	// getMatchedData: Function
-	query: string
-	matchDetails: Array<any>
+import { DetailsProps } from '../helpers/detailsdefinitions'
+import { getWeatherData } from '../helpers/apiCalls'
+// import raindrop from '../assets/liquid.png'
+// should we break this out into different components?
+// feel free to break up lines of text for readability, i'm getting tired
+const degToDirection = (deg: number):string => {
```
Comment (ErinUntermeyer) [inline src/Location/Location.tsx:21]: Thanks for spending the time to do this! Very clean. Nice job!

File: src/WeeklyData/WeeklyData.tsx [hunk 4]
```diff
@@ -0,0 +1,36 @@
+import React from 'react'
+
+
+
+const WeeklyData = (props: any) => {
```
Comment (ErinUntermeyer) [inline src/WeeklyData/WeeklyData.tsx:5]: Good call on moving this to its own file.

File: src/helpers/apiCalls.tsx [hunk 5]
```diff
@@ -1,5 +1,8 @@
-const airQualityKey = "DACE2187-D810-4B4B-81E4-45AEBAF087A0"
-const weatherDataKey = "02a1de26e83e798e7595a522e8e4e9d9"
+const airQualityKey = "DACE2187-D810-4B4B-81E4-45AEBAF087A0";
+
+const weatherDataKey = "02a1de26e83e798e7595a522e8e4e9d9";
```
Comment (ErinUntermeyer) [inline src/helpers/apiCalls.tsx:3]: Haha, I did this exact thing too!

File: src/App/App.tsx (hunk 2 above)
Comment (JoshSevy) [reply src/App/App.tsx:33]: I was having issues with the app rerendering somewhere so it was causing loops in Location, even outside of useEffect, lots of trial and error to get it work.

File: src/helpers/dataFilter.tsx [hunk 6]
```diff
@@ -0,0 +1,22 @@
+import { getCoordinates, getWeatherData, getAirQualityData} from './apiCalls';
```
Comment (ErinUntermeyer) [inline src/helpers/dataFilter.tsx:1]: The fetch call we wrote to get the details for the cards on Home could live here too I think! When we merge in, we can move it over.

**Pull request merged**